src/org/parosproxy/paros/core/scanner/ScannerParam.java

```diff
@@ -626,4 +647,7 @@ public void setAddQueryParam(boolean addQueryParam) {
         getConfig().setProperty(SCAN_ADD_QUERY_PARAM, this.addQueryParam);
     }
 
+    public String getWhitelistingValueExpression() {

```

Comment:
This should return a custom class, it hides implementation details and makes it easier to enhance in the future. It could even use `ScannerParamFilter` (by renaming `isToExclude` to `isFiltered` it would work for both cases).

Reply:
The class "ScannerParamFilter" is filterting by key and not by value isn't it? Maybe a new custom class for filtering by value would be a better choice. What do you think?

Reply:
It is now, but there's no reason it can't filter by value as well (`isToExclude` (or `isFiltered`) takes the whole parameter so which properties of the parameter it checks it's just an implementation detail). I'm fine with other class though.

Reply:
Would it be okay if i rename `ScannerParamFilter` and some other related files?
My plan is to create a central filter class which accepts specific filter implementations each related to a special topic. For example `ScannerParamFilter` changes to `ScannerParamUrlFilter` and the new value filtering feature could be named to `ScannerParamValueFilter`. The central filter class iterates over the specific implementations and returns on first match. The default return value is `false`.


 ```suggestion
public interface ScannerParamFilter {
    boolean filter(HttpMessage msg, NameValuePair param);
}

public class ScannerParamValueFilter implements ScannerParamFilter, Cloneable {
  private String whitelistingValueExpression;

  public ScannerParamValueFilter(String whitelistingValueExpression){
    this.whitelistingValueExpression = whitelistingValueExpression;
  }

  @Override public boolean filter(HttpMessage msg, NameValuePair param) {
    if (StringUtils.isEmpty(this.whitelistingValueExpression)) return false;
    return param.getValue() != null && !param.getValue().matches(this.whitelistingValueExpression);
  }
}

public class ScannerParamFilterCollection implements ScannerParamFilter {
  private List<ScannerParamFilter> filterList = new ArrayList<>();

  public void addParamFilter(ScannerParamFilter paramFilter) {
    this.filterList.add(paramFilter);
  }

  public void addParamFilters(List<? extends ScannerParamFilter> paramFilters) {
    this.filterList.addAll(paramFilters);
  }

  @Override public boolean filter(HttpMessage msg, NameValuePair param) {
    for (ScannerParamFilter paramFilter: this.filterList) {
      if (paramFilter.filter(msg, param)){
        return true;
      }
    }
    return false;
  }
}
```

Usage example:
```suggestion
        ScannerParamFilterCollection scannerParamFilterCollection = new ScannerParamFilterCollection();
        scannerParamFilterCollection.addParamFilters(excludedParameters);
        scannerParamFilterCollection.addParamFilter(getWhitelistingValueFilter());
```